### PR TITLE
Add documentation regarding how to build Ruby for debugging.

### PR DESCRIPTION
**Pull request by ioquatix**
Is this the best way @nobu? Do you have any other advice?

### CODE REVIEW

File: doc/contributing/building_ruby.md [hunk 1]
```diff
@@ -157,6 +157,14 @@ with the Ruby script you'd like to run. You can use the following make targets:
 * `make lldb-ruby`: Runs `test.rb` using Ruby in lldb
 * `make gdb-ruby`: Runs `test.rb` using Ruby in gdb
 
+### Compiling for Debugging
+
+You should configure Ruby without optimization and other flags that may interfere with debugging:
+
+``` shell
+./configure --enable-debug-env cppflags="-fno-omit-frame-pointer" optflags=-O0 LDFLAGS="-fno-omit-frame-pointer"
```
Comment (nobu) [inline doc/contributing/building_ruby.md:165]: `-fno-omit-frame-pointer` isn't a flag for `cpp`.
And `$(optflags)` is included in `LDFLAGS` not only in `CFLAGS`.

```suggestion
./configure --enable-debug-env optflags="-O0 -fno-omit-frame-pointer"
```

`--enable-devel` (which implies `debug-env`) may be worth to mention too.

Comment (ioquatix) [reply doc/contributing/building_ruby.md:165]: Ah, yes, you are totally correct.

**Comment by ioquatix** [comment]
@nobu do you think `--enable-debug-env` should set `-O0` etc?

**Comment by nobu** [comment]
Hmmm, now `optflags` is defaulted to `-O3`.
I think it's ok to replace the default.

**Comment by nobu** [comment]
> now `optflags` is defaulted to `-O3`.
if the compiler is gcc-variant.

**Comment by nobu** [comment]
Ah, `--enable-debug-env`?
It is to "enable RUBY_DEBUG environment variable", different purpose.

**Comment by ioquatix** [comment]
Thanks @nobu - I believe documentation like this is really helpful. If you have any other ideas let's discuss it.

**Pull request merged**